velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -32,6 +33,18 @@ struct ArrayMinMaxFunction {
 
   template <typename T>
   void update(T& currentValue, const T& candidateValue) {
+    if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
+      using namespace facebook::velox::util::floating_point;
+      if constexpr (isMax) {
+        if (NaNAwareGreaterThan<T>{}(candidateValue, currentValue)) {
+          currentValue = candidateValue;
+        }
+      } else {
+        if (NaNAwareLessThan<T>{}(candidateValue, currentValue)) {
+          currentValue = candidateValue;
+        }
+      }
+    }
```

Comment:
Should not we return here before proceeding on line 48?
Or 1st two conditions only tackle NaN?
Asking because not sure if below conditions can overwrite what above once did.
Maybe adding comment why we do two rounds of compare with floating point would help.